gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineClassUniquenessPlugin.java

```diff
@@ -33,8 +35,14 @@
 public class BaselineClassUniquenessPlugin extends AbstractBaselinePlugin {
     @Override
     public final void apply(Project project) {
-        TaskProvider<CheckClassUniquenessLockTask> checkClassUniqueness =
-                project.getTasks().register("checkClassUniqueness", CheckClassUniquenessLockTask.class);
+        Provider<JarClassHasher> jarClassHasher = project.getGradle()
+                .getSharedServices()
```

Comment:
The reason we use a BuildService here is to reuse the cache between tasks of different projects?

Reply:
Correct. The other alternative I'm aware of is putting an extension on the root project and locating the cache there, which should also work.